### PR TITLE
BURIED : fix explosion timer

### PR DESCRIPTION
**Pull request by kartiksharmakk**
in reference to  [Trac #13769](https://bugs.scummvm.org/ticket/13769) , during loading of global flag cgMWCatapultData with older value , there was a conflict with getMillis() which has been fixed by reinitalizing cgMWCatapultData to zero . This fixed the explosion timer , which was struck due to failing of condition `(timer + CATAPULT_TIMEOUT_VALUE < g_system->getMillis())) `

### CODE REVIEW

**Review by rvanlaar — COMMENTED** [review]
Better to comment why it needs to be reset. E.g. the explosion not happening.

**Comment by digitall** [comment]
@kartiksharmakk : Thank you for your contribution. However, two points. Firstly, your commit messages do not comply with https://wiki.scummvm.org/index.php?title=Commit_Guidelines .. They should be something like "BURIED: Fix Castle Galliard Wall Explosion Timer Bug". You should rebase and squash this to fix.

**Comment by digitall** [comment]
Secondly, while your fix will cause the explosion to occur, it is not the correct fix since it effectively just disables the following test.

As per my last comment on [Bug #13769](https://bugs.scummvm.org/ticket/13769), this bug was present in the original Win3.1 executable but fixed in later versions i.e. Win95 release. It will need someone to look at what the actual expected fix is, since I don't think just resetting that value will create the correct game logic...

**Comment by digitall** [comment]
To be clear, someone needs to look at the disassembled code around these variables in the Win 95 / patched binary with Ghidra or similar to work out what the original does.

**Comment by digitall** [comment]
> Secondly, while your fix will cause the explosion to occur, it is not the correct fix since it effectively just disables the following test.
> 
> As per my last comment on [Bug #13769](https://bugs.scummvm.org/ticket/13769), this bug was present in the original Win3.1 executable but fixed in later versions i.e. Win95 release. It will need someone to look at what the actual expected fix is, since I don't think just resetting that value will create the correct game logic...

Actually, your fix may be "the right thing", though the following test could be removed since it will always be true. However, I do have a bunch of local WIP code looking at this. It may also be possible to determine this based on original savegames since the variable is written to the savefile.

File: engines/buried/environ/castle.cpp [hunk 1]
```diff
@@ -175,17 +175,15 @@ class ExplodingWallSafeDistance : public SceneBase {
 	bool _timerStarted;
 	//uint32 _startTime;
 	bool _walkthrough;
-};
+};
```
Comment (bluegr) [inline engines/buried/environ/castle.cpp:178]:
```suggestion
};
```

**Comment by bluegr** [comment]
The proposed change looks correct, regarding the logic itself. That check for a zero value looks very fishy, and would justify the behavior described in the bug.

**Comment by bluegr** [comment]
After reviewing the proposed again, it is correct. That check for zero value was completely wrong in the constructor. Thanks again for the work on this!

**Comment by sev-** [comment]
@bluegr Did you check with the original sources before merging?

**Pull request merged**